### ISSUE TITLE
Parquet type mismatch in IN or VALUES section starting from 24.8

### ISSUE THREAD

**Issue by Selfeer** [ClickHouse/ClickHouse#69585]
We're suddenly getting this erorr when trying to insert data into a clickhouse table from a parquet file. It used to work before the 24.8 update for latest.

```
2024.09.13 03:12:10.659255 [ 3822 ] {c86a86d3-6d4b-4bf7-b08b-7517bdbb0bff} <Error> executeQuery: Code: 53. DB::Exception: Type mismatch in IN or VALUES section. Expected: DateTime. Got: Decimal64:  (When parsing Parquet statistics for column timestamp, physical type 2, column descriptor = {
  name: timestamp,
  path: timestamp,
  physical_type: INT64,
  converted_type: NONE,
  logical_type: Timestamp(isAdjustedToUTC=false, timeUnit=milliseconds, is_from_converted_type=false, force_set_converted_type=false),
  max_definition_level: 1,
  max_repetition_level: 0,
}. Please report an issue and use input_format_parquet_filter_push_down = false to work around.): (in file/uri /var/lib/clickhouse/store/de9/de99c07b-ccbe-4614-924a-3c79870a4d4d//data.Parquet): While executing ParquetBlockInputFormat: While executing File. (TYPE_MISMATCH) (version 24.3.11.7 (official build)) (from 127.0.0.1:37340) (in query: SELECT timestamp, toTypeName(timestamp) FROM table_b99a4458_716c_11ef_921b_960003b38856 FORMAT JSONEachRow ), Stack trace (when copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c8997bb
1. DB::Exception::Exception<String, DB::Field::Types::Which>(int, FormatStringHelperImpl<std::type_identity<String>::type, std::type_identity<DB::Field::Types::Which>::type>, String&&, DB::Field::Types::Which&&) @ 0x0000000010f14c95
2. DB::(anonymous namespace)::convertFieldToTypeImpl(DB::Field const&, DB::IDataType const&, DB::IDataType const*) @ 0x0000000010f0b9c3
3. DB::convertFieldToType(DB::Field const&, DB::IDataType const&, DB::IDataType const*) @ 0x0000000010f08f02
4. DB::ParquetBlockInputFormat::initializeIfNeeded() @ 0x000000001223ac3c
5. DB::ParquetBlockInputFormat::read() @ 0x000000001223dfbe
6. DB::IInputFormat::generate() @ 0x00000000120b8576
7. DB::ISource::tryGenerate() @ 0x0000000012093f55
8. DB::ISource::work() @ 0x00000000120939e2
9. DB::ExecutionThreadContext::executeTask() @ 0x00000000120aec1a
10. DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x00000000120a3170
11. DB::PipelineExecutor::executeStep(std::atomic<bool>*) @ 0x00000000120a2b88
12. DB::PullingPipelineExecutor::pull(DB::Chunk&) @ 0x00000000120b29b7
13. DB::StorageFileSource::generate() @ 0x00000000113e8510
14. DB::ISource::tryGenerate() @ 0x0000000012093f55
15. DB::ISource::work() @ 0x00000000120939e2
16. DB::ExecutionThreadContext::executeTask() @ 0x00000000120aec1a
17. DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x00000000120a3170
18. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::PipelineExecutor::spawnThreads()::$_0, void ()>>(std::__function::__policy_storage const*) @ 0x00000000120a4898
19. ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::worker(std::__list_iterator<ThreadFromGlobalPoolImpl<false>, void*>) @ 0x000000000c945e99
20. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<void ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(void&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000c9496fa
21. void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000c94850d
22. ? @ 0x00007b330ff46609
23. ? @ 0x00007b330fe6b353
```

**Comment by arthurpassos** [comment]
Does it work with `input_format_parquet_filter_push_down = false`?

**Comment by Selfeer** [comment]
It does work if you set `input_format_parquet_filter_push_down = false`. Is that a good idea to have `input_format_parquet_filter_push_down = true` as the default value when it obviously causes some issues in cases like this?

**Comment by antonio2368** [comment]
Fixed by https://github.com/ClickHouse/ClickHouse/pull/69745

**Comment by Selfeer** [comment]
@antonio2368 Thanks, the issue is fixed on the latest build but it is still there on 24.3, is there a plan to port that fix to 24.3 as well?

**Comment by antonio2368** [comment]
@Selfeer the fix was backported, but PR wasn't merged only for that version :facepalm: 
Not sure how it happened but in any case thanks for the reminder, I merged it and it should be available when we release the new patch releases.
